lib/service/service.go

```diff
@@ -2605,6 +2605,72 @@ func (process *TeleportProcess) initAuthService() error {
 		}
 	}
 
+	// We mark the process state as starting until the auth backend is ready.
+	// If cache is enabled, this will wait for the cache to be populated.
+	// We don't want auth to say its ready until its cache is populated,
+	// else a rollout might progress too quickly and cause backend pressure
+	// and outages.
+	{
+		component := teleport.Component(teleport.ComponentAuth, "backend")
+		process.ExpectService(component)
+		process.RegisterFunc("auth.wait-for-event-stream", func() error {
+			start := process.Clock.Now()
+
+			retry, err := retryutils.NewRetryV2(retryutils.RetryV2Config{
+				First:  0,
+				Driver: retryutils.NewLinearDriver(5 * time.Second),
+				Max:    time.Minute,
+				Jitter: retryutils.DefaultJitter,
+				Clock:  process.Clock,
+			})
+			if err != nil {
+				return trace.Wrap(err, "creating the backend watch retry (this is a bug)")
+			}
+			log := process.logger.With(teleport.ComponentLabel, component)
+			var attempt int
+
+			// The wait logic is retried until it is successful or a termination is triggered.
+			for {
+				attempt++
+				select {
+				case <-process.GracefulExitContext().Done():
+					return trace.Wrap(err, "context cancelled while backing off (attempt %d)", attempt)
+				case <-retry.After():
+					retry.Inc()
+				}
+
+				w, err := authServer.NewWatcher(process.ExitContext(), types.Watch{
+					Name: "auth.wait-for-backend",
+					Kinds: []types.WatchKind{
+						{Kind: types.KindClusterName},
+					},
+				})
+				if err != nil {
+					log.ErrorContext(process.ExitContext(), "Failed to create watcher", "kind", types.KindClusterName, "attempt", attempt, "error", err)
+					continue
+				}
+
+				select {
+				case evt := <-w.Events():
```

Comment:
I wonder if we can add a fallback safety timeout when it will force to start the service even if a cache is not fully healthy -  like 5m or 10m 

Imagine scenario when one of not important collection for the teleport functionality is broken like  one of the db_server element  exceeds the max gRPC message size.  

So if a bad resource is created the auth are still running but the db access flow is degraded.  If auth pods will be restarted due to troubleshooting all the flow will be broken and users will lost access to the teleport cluster entirely.